errors/errors.go

```diff
@@ -2,64 +2,73 @@ package errors
 
 import (
 	"fmt"
+	"strings"
 
 	"github.com/mondough/typhon/errors/stack"
 )
 
 type Error struct {
-	Code           int               `json:"code"`
-	Message        string            `json:"message"`
-	PublicContext  map[string]string `json:"public_context"`
-	PrivateContext map[string]string `json:"private_context"`
-	Stack          stack.Stack       `json:"stack"`
+	Code        string            `json:"code"`
+	Message     string            `json:"message"`
+	Params      map[string]string `json:"params"`
+	StackFrames stack.Stack       `json:"stack"`
 }
 
 // Generic error codes. Each of these has their own constructor for convenience.
-// You can use any integer as a code, just use the `New` method.
+// You can use any string as a code, just use the `New` method.
 const (
-	ErrUnknown         = 0
-	ErrInternalService = 1
-	ErrBadRequest      = 2
-	ErrBadResponse     = 3
-	ErrForbidden       = 4
-	ErrUnauthorized    = 5
-	ErrNotFound        = 6
-	ErrTimeout         = 7
+	ErrUnknown         = "unknown"
+	ErrInternalService = "internal_service"
+	ErrBadRequest      = "bad_request"
+	ErrBadResponse     = "bad_response"
+	ErrForbidden       = "forbidden"
+	ErrUnauthorized    = "unauthorized"
+	ErrNotFound        = "not_found"
+	ErrTimeout         = "timeout"
 )
 
-// Error returns a string message of the error
+// Error returns a string message of the error. It is a concatenation of Code and Message params
 // This means the Error implements the error interface
 func (p *Error) Error() string {
 	if p == nil {
 		return ""
 	}
-	return p.Message
+
+	return fmt.Sprintf("%s: %s", p.Code, p.Message)
 }
 
 // StackString formats the stack as a beautiful string with newlines
 func (p *Error) StackString() string {
 	stackStr := ""
-	for _, frame := range p.Stack {
+	for _, frame := range p.StackFrames {
 		stackStr = fmt.Sprintf("%s\n  %s:%d in %s", stackStr, frame.Filename, frame.Line, frame.Method)
 	}
 	return stackStr
 }
 
-// VerboseString returns the error message, stack trace and contexts
+// VerboseString returns the error message, stack trace and params
 func (p *Error) VerboseString() string {
-	return fmt.Sprintf("%s\nPublic context: %+v\nPrivate context: %+v\n%s", p.Error(), p.PublicContext, p.PrivateContext, p.StackString())
+	return fmt.Sprintf("%s. %s\nParams: %+v\n%s", p.Error(), p.Message, p.Params, p.StackString())
```

Comment:
If I'm reading this right, the output will be `dotted.code: message. message\nstack`